### PR TITLE
Text updates based on the review by Bob Briscoe

### PR DESCRIPTION
**Pull request by hannestschofenig**
First chunk of changes.

### CODE REVIEW

File: draft-ietf-suit-architecture.md [hunk 1]
```diff
@@ -70,63 +71,124 @@ informative:
...
 as well as adding new functionality is recommended by security
 experts.
 
-This document lists requirements and describes an architecture for
-a firmware update mechanism suitable for IoT devices. The
-architecture is agnostic to the transport of the firmware images
-and associated meta-data.
+In addition to the definition of terminology and an architecture
+this document motivates the standardization of a manifest format
+as transport-agnostic means for describing and protecting firmware
```
Comment (oyvindronningstad) [inline draft-ietf-suit-architecture.md]: Not 100% sure, but
```suggestion
as a transport-agnostic means for describing and protecting firmware 
```

File: draft-ietf-suit-architecture.md [hunk 2]
```diff
@@ -70,63 +71,124 @@ informative:
...
-intervention is cost prohibitive or otherwise difficult. Updates
-to the firmware of an IoT device are done to fix bugs in software,
-to add new functionality, and to re-configure the device to work
-in new environments or to behave differently in an already
-deployed context.
-
-The firmware update process, among other goals, has to ensure that
+Firmware updates can help to fix security vulnerabilities and are 
+considered to be an important building block in securing IoT devices. 
+Due to rising concerns about insecurre IoT devices the Internet
```
Comment (oyvindronningstad) [inline draft-ietf-suit-architecture.md]:
```suggestion
Due to rising concerns about insecure IoT devices the Internet
```

File: draft-ietf-suit-architecture.md [hunk 3]
```diff
@@ -70,63 +71,124 @@ informative:
...
+the bigger picture. A report about this workshop can be found at
+{{RFC8240}}. The workshop revealed a number of challenges for developers
+and led to the formation of the IETF 
+Software Updates for Internet of Things (SUIT) working group.
+
+Developing secure Internet of Things (IoT) devices is not an easy 
+task and supporting a firmware update solution requires skillful 
+engineers. Once devices are deployed, firmware updates play a 
+critical part in their lifecycle management, particularly when devices have a 
+long lifetime, are deployed in remote or inaccessible areas where manual
```
Comment (oyvindronningstad) [inline draft-ietf-suit-architecture.md]:
```suggestion
long lifetime, or are deployed in remote or inaccessible areas where manual
```

File: draft-ietf-suit-architecture.md [hunk 4]
```diff
@@ -70,63 +71,124 @@ informative:
...
+intervention is cost prohibitive or otherwise difficult. Firmware updates  
+for IoT devices are expected to work automatic, i.e. without user
```
Comment (oyvindronningstad) [inline draft-ietf-suit-architecture.md]:
```suggestion
for IoT devices are expected to work automatically, i.e. without user 
```

File: draft-ietf-suit-architecture.md [hunk 5]
```diff
@@ -70,63 +71,124 @@ informative:
...
+engineers. Once devices are deployed, firmware updates play a 
+critical part in their lifecycle management, particularly when devices have a 
+long lifetime, are deployed in remote or inaccessible areas where manual
+intervention is cost prohibitive or otherwise difficult. Firmware updates  
+for IoT devices are expected to work automatic, i.e. without user 
+involvement. Automatic updates that do not require human intervention 
+are key to a scalable solution for fixing software vulnerabilities.
+
+Firmware updates are not only done to fix 
+bugs, but they can also add new functionality, and to re-configure
```
Comment (oyvindronningstad) [inline draft-ietf-suit-architecture.md]:
```suggestion
bugs, but they can also add new functionality, and re-configure
```

File: draft-ietf-suit-architecture.md [hunk 6]
```diff
@@ -70,63 +71,124 @@ informative:
...
 While the standardization work has been informed by and optimised for firmware
 update use cases of Class 1 devices (according to the device class
 definitions in RFC 7228 {{RFC7228}}) devices, there is nothing in
 the architecture that restricts its use to only these constrained IoT devices.
 Moreover, this architecture is not limited to managing software updates,
 but can also be applied to managing the delivery of arbitrary data, such
-as configuration information and keys.
+as configuration information and keys. Unlike higher end devices, like 
+laptops and desktop PCs, many IoT devices do not have user interfaces 
+and support for unattended updates are therefore essential for the design
```
Comment (oyvindronningstad) [inline draft-ietf-suit-architecture.md]:
```suggestion
laptops and desktop PCs, many IoT devices do not have user interfaces. 
Support for unattended updates are therefore essential for the design
```

**Review by oyvindronningstad — COMMENTED** [review]
Quick partial language review

File: draft-ietf-suit-architecture.md [hunk 7]
```diff
@@ -541,258 +503,147 @@ the download process.  When a download is server-initiated,
...
-+--------+                  +-----------+                   +--------+
-     ^                                                          *
-     *                                                          *
-     ************************************************************
-                        End-to-End Security
-~~~~
-{: #e2e-figure title="End-to-End Security."}
+One way to achieve this functionality is to provide a minimum of two storage 
+locations for firmware. An alternative approach is to use a second stage 
+bootloader with build-in full featured firmware update functionality such
```
Comment (bremoran) [inline draft-ietf-suit-architecture.md]: bootloader with built-in full featured firmware update functionality such

File: draft-ietf-suit-architecture.md [hunk 8]
```diff
@@ -844,180 +695,200 @@ to be upgraded atomically.  Because the updates are happening on
...
--  to (optionally) offer attestation information (such as
-   measurements).
-
-While the software architecture of the bootloader and its
-security mechanisms are implementation-specific, the manifest can
-be used to control the firmware download from the Internet in
-addition to augmenting secure boot process. These building blocks
-are highly relevant for the design of the manifest.
+Using firmware updates to fix vulnerabilities in devices is important but securing this 
+update mechanism is equally important since security problems are exacerbated. Failure
```
Comment (bremoran) [inline draft-ietf-suit-architecture.md]: > exacerbated by the update mechanism: update is essentially authorised remote code execution, so any security problems in the update process expose that remote code execution system.

**Review by bremoran — COMMENTED** [review]
This is a really great set of edits. I think that it really improves the document as a whole! I have suggested a couple of minor edits.

**Pull request merged**